observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemRegistrar.java

```diff
@@ -187,10 +187,11 @@ public Predicate<String> getSubsystemFilter() {
         }, OperationContext.Stage.VERIFY);
 
         return CapabilityServiceInstaller.builder(MICROMETER_COLLECTOR_RUNTIME_CAPABILITY, collectorSupplier)
-                .requires(List.of(mccf, executor, processStateNotifier))
-                .withCaptor(captor::set) // capture the provided value
-                .asActive() // Start actively
-                .build();
+            .requires(List.of(mccf, executor, processStateNotifier))
+            .withCaptor(captor::set) // capture the provided value
+            .onRemove(wildFlyRegistry::close)
```

Comment:
@jasondlee I am afraid is unfortunately not correct. The service supplies a MicrometerCollector but it's closing a WildFlyRegistry. So, it's the MicrometerCollector that should also implement AutoCloseable and do proper cleanup, among other things closing the WildFlyRegistry.